### PR TITLE
Karim2629 patch 3

### PR DESCRIPTION
**Pull request by karim2629**
## Description

[Describe the changes that you've made in this pull request and explain why they are necessary.]

## Checklist

Please ensure that your pull request meets the following requirements:

- [ ] Code compiles without errors and warnings
- [ ] Code follows the project's coding style guidelines
- [ ] Code is tested and test cases pass
- [ ] Code has been reviewed by at least one other developer

## Related Issues

[If your pull request is related to any issues, link them here.]

## Additional Notes

[Include any additional information or notes that may be helpful in reviewing your pull request.]

### CODE REVIEW

**Review by karim2629 — COMMENTED** [review]
updated sort class to be able to pass file names as arguments; sorter.Reduce("input.txt", "out.txt")

File: Sort.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+using std::endl;
+using std::ifstream;
+using std::map;
+using std::ofstream;
+using std::pair;
+using std::sort;
+using std::string;
+using std::vector;
+
+void Sort::Reduce(const string& input_filename, const string& output_filename) {
```
Comment (vnijat93) [inline Sort.cpp:20]: Rename this function to `sort`

File: Sort.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    ofstream outfile(output_filename);
```
Comment (vnijat93) [inline Sort.cpp:21]: use `FileManager` to read from files

File: Sort.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#ifndef SORT_H
+#define SORT_H
+
+#include <map>
+#include <string>
+#include <vector>
+
+class Sort {
+public:
+    void Reduce(const std::string& input_filename, const std::string& output_filename);
```
Comment (vnijat93) [inline Sort.h:10]: update the method name here

File: Sort.cpp [hunk 4]
```diff
@@ -0,0 +1,83 @@
+#include "Sort.h"
+#include <algorithm>
+#include <fstream>
```
Comment (mcockle23) [inline Sort.cpp:3]: you shouldn't need fstream or iostream since we will use FileManager for file reads and writes

File: Sort.cpp [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+string Sort::ExtractString(const string& pair_string) {
+    size_t start_index = pair_string.find_first_of("\"") + 1;
+    size_t end_index = pair_string.find_last_of("\"");
+    return pair_string.substr(start_index, end_index - start_index);
+}
+
+bool Sort::UpdateWordCounts(const string& input_filename, map<string, vector<int>>& word_counts) {
+    ifstream infile(input_filename);
```
Comment (mcockle23) [inline Sort.cpp:48]: update this to FileManager calls

**Review by mcockle23 — CHANGES_REQUESTED** [review]

**Review by mcockle23 — APPROVED** [review]

**Pull request merged**